index.js

```diff
@@ -1,4 +1,6 @@
 require('rootpath')();
 require('@babel/register');
 
+require('dotenv').config({ path: '.env' });
```

Comment:
> More fundamentally, `dotenv` expects a syntax like `FOO=bar` whereas our `.env` files are formatted to export variables (i.e. `export SET FOO=bar`). See [the example](https://github.com/NYPL/discovery-front-end/blob/development/.env-sample). So if we wanted to switch to using `dotenv`, we'd have to also change that practice..

You've brought up a good point regarding the syntax differences in the .env file. This was not considered. I'll need to review that. 

However, referring to your first question 
> Is this going to error if there is no `.env` on the server? There is no `.env` on the server and we should never put one in source control, to be clear. So I'm wondering if deployed code will fail when there is no `.env`.
 
This will not conflict. If a file does not exist, no error is thrown. A returned error object is returned from `dotenv.config` which we can choose to do with as we please, but it will not conflict with any process or initialization.

Reply:
Ah - I see now what you're saying.. It's an error "object" but not a thrown error. Got it! thanks

Reply:
@nonword 
I would suggest we leave in the ReadMe `source .env;` and extend this with an option for the dot env approach. 

The dotenv package will not process any environment variables that have `export set ...` 
An empty object is returned and nothing is added to the `process.env`. 

There are two options of setting up an environment file and running a local development instance.
1. `source .env; npm start` - Works for *nix OS
a. Ex .env: `export set [key]=[value]`
b. If envs are set using `[key]=[value]` dotenv will add them to the process rendering `source .env` superfluous 

2. `npm start` - Works for Windows OS && *nix OS
a. Ex .env: `[key]=[value]`
b. Environment variables will not be set if configured as `export set [key]=[value]`

^ This can be added to the ReadMe for clarity.